src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -893,15 +893,35 @@ def test_process_array_declarations():
     reader = FortranStringReader("integer :: l5(2), l6(3)")
     fparser2spec = Specification_Part(reader).content[0]
     processor.process_declarations(fake_parent, [fparser2spec], [])
-    assert fake_parent.symbol_table.lookup("l5").datatype.shape == [2]
```

Comment:
It's a pity we can not compare shapes in this way (`datatype.shape == [2]`) anymore... (unless we want to use the dark path of redefining `__eq__` )

Reply:
I was wondering whether we could have our own `same_as()`, or `equivalent()` function for Literals. That would allow us to be able to configure whether we care if things are different from each other or not e.g. we would want datatype to be the same but might not care about precision so could have an `ignore_precision` optional argument.

Reply:
I like the `same_as()` with optional `ignore_precision` idea, but since this is a different functionality than the focus of this PR maybe you can push it to another issue and see what other people think.

Reply:
Yes, I wasn't suggesting it for this PR :-). It would be better to have a `same_as()` or equivalent method for a `DataType` rather than a `Literal`, a least as a starting point. In addition we could think of having a `same_as` for all `DataNode`s rather than just a `Literal` which could have `value`, `datatype` and `precision` as optional arguments. Anyway, I've captured this in issue #1016